server/src/main/java/org/elasticsearch/monitor/StatusInfo.java

```diff
@@ -35,6 +37,20 @@ public Status getStatus() {
         return status;
     }
 
+    private static Status readStatus(StreamInput in) throws IOException {
+        String statusString = in.readOptionalString();
+        if (statusString == null) {
+            return null;
+        }
+        return Status.valueOf(statusString);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalString(status == null ? null : status.name());
+        out.writeOptionalString(info);
```

Comment:
Are we changing the semantics for `StatusInfo` ? I don't think these fields are optional now. I think we should keep them non-null (ie. mandatory).

Reply:
There was nothing preventing them from being null. I thought it was very unlikely that someone would pass in nulls (what would be the point?). But if they do, I don't want it to blow up during serialization. From what I can tell there's no good way to enforce null checks on a record. What do you think?

Reply:
I caught up with @dakrone on this and we were thinking of aligning the object to the requirements (ie. non-null fields) via `Objects.requireNonNull` calls in the constructor, alongside strict (non-optional) serialisation.

What do you think @masseyke ?

Reply:
I'm all for enforcing non-nullable fields. I'll change the serialization to assume no nulls, and we can enforce it on a subsequent PR.